### PR TITLE
Integration Test Cases for Unsupported TF cases

### PR DESCRIPTION
**Pull request by hawflau**
#### Why is this change necessary?
Add integration test cases for unsupported TF cases

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/integration/buildcmd/test_build_terraform_applications.py [hunk 1]
```diff
@@ -432,3 +439,40 @@ def test_build_and_invoke_lambda_functions(self, function_identifier):
...
+    "Skip Terraform test cases unless running in CI and not in Windows",
+)
+class TestUnsupportedCases(BuildTerraformApplicationIntegBase):
+    terraform_application = Path("terraform/unsupported")
+
+    @parameterized.expand(
+        [
+            (
+                "conditional_layers",
+                r"SAM CLI could not process a Terraform project that contains Lambda functions that are linked to more than one lambda layer",
```
Comment (sriram-mv) [inline tests/integration/buildcmd/test_build_terraform_applications.py:455]: we should change this error message to say "AWS SAM CLI"

File: tests/integration/buildcmd/test_build_terraform_applications.py [hunk 2]
```diff
@@ -432,3 +439,40 @@ def test_build_and_invoke_lambda_functions(self, function_identifier):
                 "multiValueHeaders": None,
             },
         )
+
+
+@skipIf(
+    not (CI_OVERRIDE and not IS_WINDOWS),
```
Comment (sriram-mv) [inline tests/integration/buildcmd/test_build_terraform_applications.py:445]: how do we ensure we run these tests on windows as well?

**Review by sriram-mv — COMMENTED** [review]
Overall: looks okay, I did not go too deep into the examples themselves.

File: tests/integration/buildcmd/test_build_terraform_applications.py (hunk 1 above)
Comment (hawflau) [reply tests/integration/buildcmd/test_build_terraform_applications.py:455]: sure, will update the error message.

**Review by moelasmar — APPROVED** [review]
it looks good to me, thanks Wing

**Review by sriram-mv — APPROVED** [review]

**Pull request merged**